## Example:

In [1]:
import pandas as pd

data = {
    'AAPL': [0.01, 0.02, -0.01, 0.00, 0.03],
    'GOOG': [0.02, 0.01, -0.02, 0.01, 0.04],
    'TSLA': [-0.01, 0.00, 0.01, -0.02, 0.05]
}

df = pd.DataFrame(data)

correlation = df.corr()
print(correlation)

from sklearn.linear_model import LinearRegression

X = df[['GOOG', 'TSLA']]  # predictors
y = df['AAPL']            # target

model = LinearRegression().fit(X, y)

print("Intercept:", model.intercept_)
print("Coefficients:", model.coef_)

          AAPL      GOOG      TSLA
AAPL  1.000000  0.875190  0.585206
GOOG  0.875190  1.000000  0.443877
TSLA  0.585206  0.443877  1.000000
Intercept: 0.002431941923774954
Coefficients: [0.55898367 0.14337568]


- Correlation says: “Do Apple and Google move together?”
- Regression says: “How much does Apple tend to move when Google or Tesla move?

A 1% increase in GOOG's return will yield in a 0.55% of AAPL, and a 1% increase in TSLA will yield 0.14% in AAPL.

In [2]:
df

AAPL  GOOG  TSLA
0  0.01  0.02 -0.01
1  0.02  0.01  0.00
2 -0.01 -0.02  0.01
3  0.00  0.01 -0.02
4  0.03  0.04  0.05

## The Real Deal:

In [3]:
import yfinance as yf
tickers_list = ['MSFT', 'GOOG']
data = yf.download(tickers_list, start='2020-01-01', end='2021-01-01', interval = '1mo')['Close']

df = pd.DataFrame(data)
data

C:\Users\kevin\AppData\Local\Temp\ipykernel_16312\722445306.py:3: FutureWarning: YF.download() has changed argument auto_adjust default to True
  data = yf.download(tickers_list, start='2020-01-01', end='2021-01-01', interval = '1mo')['Close']
[*********************100%***********************]  2 of 2 completed


Ticker           GOOG        MSFT
Date                             
2020-01-01  71.287758  162.198914
2020-02-01  66.570801  154.366714
2020-03-01  57.796947  150.680038
2020-04-01  67.034538  171.221664
2020-05-01  71.023827  175.081558
2020-06-01  70.262840  194.980011
2020-07-01  73.709854  196.417130
2020-08-01  81.226173  216.077087
2020-09-01  73.045807  202.001266
2020-10-01  80.571571  194.452530
2020-11-01  87.516785  205.593201
2020-12-01  87.076401  214.171814

In [6]:
correlation = df.corr()
print(correlation)

from sklearn.linear_model import LinearRegression

X = df[['MSFT']]  # predictors
y = df['GOOG']            # target

model = LinearRegression().fit(X, y)

print("Intercept:", model.intercept_)
print("Coefficients:", model.coef_)

Ticker      GOOG      MSFT
Ticker                    
GOOG    1.000000  0.844058
MSFT    0.844058  1.000000
Intercept: 13.54337947637007
Coefficients: [0.32388216]


1% increase in MSFT will result in 0.32% increase of GOOG 